Bottom: 1c44f65785350dbe3d32501005f2af6aeebebb92
Top:    7039b3e2c1d4f06bb2b8d8f467d0e14500e4ef08
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-04 00:38:31 +0000

Play with perspective rendering


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index edb7498..cf76ee7 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1710,13 +1710,26 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
 
   glMatrixMode (GL_PROJECTION);
   glLoadIdentity ();
-  glOrtho (0, widget->allocation.width, widget->allocation.height, 0, -100000, 100000);
+//  glOrtho (0, widget->allocation.width, widget->allocation.height, 0, -100000, 100000);
+//  glOrtho (-1, 1, -1, 1, 100, 10000);
+//  glFrustum (0, widget->allocation.width, widget->allocation.height, 0, 10, 1000);
+  glFrustum (-1, 1, -1, 1, 10, 10000);
+//  gluPerspective (30, widget->allocation.width / widget->allocation.height, 10, 1000);
   glMatrixMode (GL_MODELVIEW);
   glLoadIdentity ();
 
+  glScalef (2, 2, 1);
+
+  glTranslatef (-1., 1., 0.);
+  glScalef ( 2. / (float)widget->allocation.width,
+            -2. / (float)widget->allocation.height,
+             1. / 100.);
+
+  glTranslatef (0., 0., -2000.);
   glTranslatef (widget->allocation.width / 2., widget->allocation.height / 2., 0);
   glMultMatrixf ((GLfloat *)view_matrix);
   glTranslatef (-widget->allocation.width / 2., -widget->allocation.height / 2., 0);
+  glScalef (1., 1., 1. / 3.);
   glGetFloatv (GL_MODELVIEW_MATRIX, (GLfloat *)last_modelview_matrix);
 
 #if 0
